coderd/tracing/httpmw.go

```diff
@@ -24,7 +25,10 @@ func HTTPMW(tracerProvider *sdktrace.TracerProvider, name string) func(http.Hand
 			defer span.End()
 			r = r.WithContext(ctx)
 
-			wrw := middleware.NewWrapResponseWriter(rw, r.ProtoMajor)
+			sw, ok := rw.(*httpapi.StatusWriter)
+			if !ok {
+				panic("ResponseWriter not a *httpapi.StatusWriter?")
```

Comment:
Suggestion: Should we include the type for debuggability?

```suggestion
				panic(fmt.Sprintf("ResponseWriter not a *httpapi.StatusWriter; got %T", rw))
```